gameplay.py

```diff
@@ -112,12 +209,28 @@ def use_move(self, attacking_pokemon: Pokemon, defending_pokemon: Pokemon,
         total_dmg_dealt = sum(hit_damages) // self.LENGTH_MODIFIER
 
         attacker_health_delta = math.floor(move_used.info.drain * total_dmg_dealt)
+        recoil = attacker_health_delta < 0
+        # Apply damage from status ailments
+        attacker_health_delta -= attacking_pokemon.get_status_damage(
+            total_dmg_dealt > defending_pokemon.hp
+        )
+
+        if move_used.info.name == "attack_self":
+            print(f"{attacking_pokemon.nickname} is confused! It hurt itself in its confusion!")
+            attacking_pokemon.apply_health_effect(attacker_health_delta)
+            return
 
         print(f"{attacking_pokemon.nickname} dealt {total_dmg_dealt} damage{' in total' if len(hit_damages) > 1 else ''}.")
-        if attacker_health_delta != 0:
-            attacker_action_text = f"healed itself for {attacker_health_delta}" \
-                if attacker_health_delta > 0 else f"was hit with {attacker_health_delta} recoil damage"
-            print(f"{attacking_pokemon.nickname} {attacker_action_text}")
+        if attacker_health_delta > 0:
+            print(f"{attacking_pokemon.nickname} healed itself for {attacker_health_delta}")
+        elif attacker_health_delta < 0 and recoil:
+            print(f"{attacking_pokemon.nickname} was hit with {attacker_health_delta} recoil damage")
+
+        if move_used.info.ailment:
+            ailment_chance = move_used.info.ailment_chance
+            # Moves that always apply an ailment have an `ailment_chance` of 0 for some reason
```

Comment:
See comment below about mapping a 0% ailment chance to 100% at scrape-time vs checking it here.